### PR TITLE
Use Quarkus Platform BOM

### LINKED ISSUE #11015: Use dist Quarkus version in the operator
For the operator, we're currently [overriding](https://github.com/keycloak/keycloak/blob/8cd2c97982847b5bf322afb3423e3efbb7fa8fdb/operator/pom.xml#L36) the Quarkus version that is [set at project level](https://github.com/keycloak/keycloak/blob/8cd2c97982847b5bf322afb3423e3efbb7fa8fdb/pom.xml#L44) and used by the Quarkus dist. We should unify it and use a single version without any overrides.

The risk is potential incompatibilities with JOSDK that might bring fast Quarkus version updates.

### LINKED ISSUE #15870: Cleanup unused WildFly dependencies.
### Description

After the Quarkus based distribution became the new default, we should check, which WildFly dependencies remained unused and we should do a proper cleanup.

### LINKED ISSUE #20570: Use Quarkus Platform BOMs
### Description

To further align with Quarkus, we should be using the Platform BOMs. In particular:
* [`io.quarkus.platform:quarkus-bom`](https://github.com/quarkusio/quarkus-platform/blob/main/generated-platform-project/quarkus/bom/pom.xml)
* [`io.quarkus.platform:quarkus-operator-sdk-bom`](https://github.com/quarkusio/quarkus-platform/blob/main/generated-platform-project/quarkus-operator-sdk/bom/pom.xml). This in particular is important as we won't be no longer specifying an Operator SDK version but rather using the version that comes with a certain Quarkus version.

### PR DESCRIPTION
**Pull request by Pepo48**
* added Quarkus Platform BOM
* added Quarkus Operator SDK BOM
* removed WildFly dependencies from the Quarkus module
* removed all the dependency overrides from the parent POM

Closes #20570
Closes #15870

Signed-off-by: Peter Zaoral <pzaoral@redhat.com>

I might've missed dependencies that can be removed as a part of this PR. Therefore, please, let me know if you find such during the review. Thanks!

### CODE REVIEW

File: operator/pom.xml
Comment (Pepo48) [reply operator/pom.xml]: @mabartos I actually found out that OLM CI workflow is failing due to this change (see the comparison between [this](https://github.com/Pepo48/keycloak/actions/runs/5332061993) and [this](https://github.com/Pepo48/keycloak/actions/runs/5332663933)). So far I haven't found the connection between the property existence and the operator deployment, but I haven't investigated it properly just yet.

Does it ring any bell right away? I plan to test it on a local cluster tomorrow.

**Review by ASzc — APPROVED** [review]
Seems ok from a product perspective. The runtime/buildtime version property split has been kept, so that's good

**Review by vmuzikar — COMMENTED** [review]
@Pepo48 Thanks for the PR.

Part of the effort around using Quarkus BOM is to stop specifying versions of the dependencies that are already present in the BOM. I think there are lots of them in the root POM. Do you plan to address it in this PR, or will it be a follow-up? Either way works for me. We need to be careful now about not doing anything too dangerous now, just days before release.

File: operator/pom.xml
Comment (vmuzikar) [reply operator/pom.xml]: It does ring a bell. We're [overriding the version in the root POM](https://github.com/keycloak/keycloak/blob/56db0de705f21f71690f915eb42905a1c56c40d4/pom.xml#L134) to something that's incompatible with Quarkus. We should remove all the overrides we don't need, see my comment below.

Comment (Pepo48) [reply operator/pom.xml]: I think it's resolved.

**Comment by Pepo48** [comment]
> @Pepo48 Thanks for the PR.
> 
> Part of the effort around using Quarkus BOM is to stop specifying versions of the dependencies that are already present in the BOM. I think there are lots of them in the root POM. Do you plan to address it in this PR, or will it be a follow-up? Either way works for me. We need to be careful now about not doing anything too dangerous now, just days before release.

I am working on it in [a separate branch](https://github.com/Pepo48/keycloak/tree/platform-bom-test), the intersection is quite big (at least based on my analysis from two days ago it's not just couple of dependencies). I'm testing it continuously, I didn't want to clutter this PR with unnecessary force pushes.

And yes, my original intention was to keep it separate just because of the size. But I can easily merge it, it doesn't matter to me much. I will finish the rest of the cleanup tomorrow.

File: operator/pom.xml
Comment (vmuzikar) [reply operator/pom.xml]: @Pepo48 Works for me but we'll need a blessing from the core team at minimum once this PR is finalized.

**Comment by Pepo48** [comment]
### What was removed from the parent BOM:
```
aesh - bumped from 2.4 to 2.7

bc-fips
bctls-fips

commons-io
commons-lang3

hibernate-core

httpclient
httpcore
httpmime

jackson-annotations
jackson-corex
jackson-databindx
jackson-dataformat-cbor
jackson-datatype-jdk8
jackson-jaxrs-base
jackson-jaxrs-json-provider
jackson-module-jaxb-annotations

jakarta.activation-api
jakarta.persistence-api
jakarta.servlet-api
jakarta.transaction-api
jakarta.ws.rs-api
jakarta.xml.bind-api

javaparser-core - bumped from 3.24.2 to 3.25.2
javax.annotation-api

jboss-logging-annotations
jboss-logging-processor
jboss-logging

jna - bumped from 4.5.1 to 5.8.0
liquibase-core
microprofile-metrics-api - bumped from 3.0.1 to 4.0.1
slf4j-api
```

### What can be pottentially removed (and for now I left untouched):
`bcpkix-jdk15on` - `bcpkix-jdk18on` is available in the platform BOM
`bcprov-jdk15on` - `bcprov-jdk18on` is available in the platform BOM

`guava` - we currently use a newer version compared to the one available in the platform BOM - 31.1
infinispan -  we currently use a newer version compared to the one available in the platform BOM - 14.0.9

`h2`- we use the same version as the BOM provides, but based on the h2.version, the jdbc driver version is determined too, for now I leave the h2 dependency untouched in order to prevent a version mismatch in the future. The same seems to apply to mysql-connector-java.

### Things I would like to discuss before I address them:

- should we explicitely exclude all the *keycloak* dependencies from the platform BOM or we don't care since everything is overriden anyway?
- can dependencies with excluded transitive dependencies be removed as well? do we care about the exclusions on the parent POM level at this point (see e.g. `liquibase-core` or `resteasy` dependencies)
- what about the dependencies that have a scope specified on the parent POM level? (see e.g `protostream-processor`)

**Comment by Pepo48** [comment]
jFTR, https://github.com/keycloak/keycloak/pull/21352 will (most likely*) affect the dependency list above in a following way:
```
bctls-fips - bumped from 1.0.14.1 to 1.0.16
commons-io - bumped from 2.11.0 to 2.13
javaparser-core - bumped from 3.24.2 to 3.25.3
```

* I compared it with 3.2.0.CR1, because the final platform BOMs were not released just yet.

**Comment by vmuzikar** [comment]
This is currently waiting for Quarkus Platform BOMs 3.2.0.Final to be released.

**Comment by vmuzikar** [comment]
This was superseded by https://github.com/keycloak/keycloak/pull/21426 (@Pepo48 is on PTO)

**Pull request closed without merging**
